common/src/types/tasks.rs

```diff
@@ -319,6 +320,8 @@ pub struct UpdateWalletTaskDescriptor {
     pub timestamp_received: u64,
     /// The external transfer, if one exists
     pub external_transfer: Option<ExternalTransfer>,
+    /// The auxiliary data for the external transfer, if one exists
```

Comment:
might be cleaner to build `ExternalTransferWithAuthData` or similar. These two fields should always either both be `Some` or `None`. 

We can also add constructors for this like `ExternalTransferWithAuthData::deposit` (enforcing the direction to be a deposit) and `ExternalTransferWithAuthData::withdrawal`